From 10791e604015ef999e5f185c82d9535625c25d9f Mon Sep 17 00:00:00 2001
From: Alexander Born <alexander.born@bmw.de>
Date: Tue, 21 Apr 2020 10:56:10 +0200
Subject: [PATCH 1/4] fix compile flags

---
 aspects.bzl | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/aspects.bzl b/aspects.bzl
index e0c86e8..af1580e 100644
--- a/aspects.bzl
+++ b/aspects.bzl
@@ -37,6 +37,7 @@ _cpp_extensions = [
     "cc",
     "cpp",
     "cxx",
+    "h",
 ]
 
 _cc_rules = [
@@ -271,8 +272,8 @@ def _compilation_database_aspect_impl(target, ctx):
     compile_flags += [
         # Use -I to indicate that we want to keep the normal position in the system include chain.
         # See https://github.com/grailbio/bazel-compilation-database/issues/36#issuecomment-531971361.
-        "-I " + str(d)
-        for d in cc_toolchain.built_in_include_directories
+        # "-I " + str(d)
+        # for d in cc_toolchain.built_in_include_directories
     ]
     compile_command = compiler_info.compiler + " " + " ".join(compile_flags) + compiler_info.force_language_mode_option
 
-- 
2.17.1


From 987af0fee2d093d2b299638229d8c6ec43754700 Mon Sep 17 00:00:00 2001
From: Alexander Born <alexander.born@bmw.de>
Date: Tue, 21 Apr 2020 11:54:43 +0200
Subject: [PATCH 2/4] fix config

---
 generate.sh | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/generate.sh b/generate.sh
index f110158..9266e82 100755
--- a/generate.sh
+++ b/generate.sh
@@ -30,13 +30,6 @@ usage() {
   printf "\n"
 }
 
-while getopts "sh" opt; do
-  case "${opt}" in
-    "s") source_dir=1 ;;
-    "h") usage; exit 0;;
-    *) >&2 echo "invalid option ${opt}"; exit 1;;
-  esac
-done
 
 # This function is copied from https://source.bazel.build/bazel/+/master:scripts/packages/bazel.sh.
 # `readlink -f` that works on OSX too.
-- 
2.17.1


From 3662c970fec738d803070b885ee94f5f583f7874 Mon Sep 17 00:00:00 2001
From: Alexander Born <alexander.born@bmw.de>
Date: Tue, 21 Apr 2020 11:55:01 +0200
Subject: [PATCH 3/4] only certain targets

---
 generate.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/generate.sh b/generate.sh
index 9266e82..76335ec 100755
--- a/generate.sh
+++ b/generate.sh
@@ -83,7 +83,7 @@ readonly QUERY_CMD=(
   bazel query
     --noshow_progress
     --noshow_loading_progress
-    'kind("cc_(library|binary|test|inc_library|proto_library)", //...) union kind("objc_(library|binary|test)", //...)'
+    'kind("cc_(library|binary|test|inc_library|proto_library)", //application/adp/activities/... + //application/adp/perception/... + //application/adp/bmw/reprocessing/... + //application/adp/common/...)'
 )
 
 # Clean any previously generated files.
-- 
2.17.1


From 34970eeba77d4a345e9091e25f1dc7d44eaa0c99 Mon Sep 17 00:00:00 2001
From: Alexander Born <alexander.born@bmw.de>
Date: Wed, 22 Apr 2020 12:26:52 +0200
Subject: [PATCH 4/4] fix compiler specific command line arguments

---
 aspects.bzl | 4 ++++
 generate.sh | 1 +
 2 files changed, 5 insertions(+)

diff --git a/aspects.bzl b/aspects.bzl
index af1580e..1ada34a 100644
--- a/aspects.bzl
+++ b/aspects.bzl
@@ -152,6 +152,8 @@ def _cc_compiler_info(ctx, target, srcs, feature_configuration, cc_toolchain):
                      get_compile_flags(target) +
                      (ctx.rule.attr.copts if "copts" in dir(ctx.rule.attr) else []))
 
+    force_language_mode_option = force_language_mode_option + " -Wno-unknown-argument -Wno-unknown-warning-option"
+
     return struct(
         compile_variables = compile_variables,
         compiler_options = compiler_options,
@@ -233,6 +235,8 @@ def _objc_compiler_info(ctx, target, srcs, feature_configuration, cc_toolchain):
                      frameworks +
                      (ctx.rule.attr.copts if "copts" in dir(ctx.rule.attr) else []))
 
+    force_language_mode_option = force_language_mode_option + " -Qunused-arguments -Wno-unknown-argument -Wno-unknown-warning-option"
+
     return struct(
         compile_variables = compile_variables,
         compiler_options = compiler_options,
diff --git a/generate.sh b/generate.sh
index 76335ec..0dd4b80 100755
--- a/generate.sh
+++ b/generate.sh
@@ -121,6 +121,7 @@ else
   ln -f -s "${WORKSPACE}/${COMPDB_FILE}" "${EXEC_ROOT}/"
 fi
 sed -i.bak -e "s|-isysroot __BAZEL_XCODE_SDKROOT__||" "${COMPDB_FILE}"  # Replace -isysroot __BAZEL_XCODE_SDKROOT__ marker
+sed -i "s/ -fno-canonical-system-headers//g" "${COMPDB_FILE}"
 
 # Clean up backup file left behind by sed.
 rm "${COMPDB_FILE}.bak"
-- 
2.17.1

